code/unit_tests/map_tests.dm

```diff
@@ -0,0 +1,99 @@
+/datum/unit_test/apc_area_test
+	name = "MAP: Area Test APC / Scrubbers / Vents Z level 1"
+
+/datum/unit_test/apc_area_test/start_test()
+	var/list/bad_areas = list()
+	var/area_test_count = 0
+	var/list/exempt_areas = typesof(/area/space,
+					/area/syndicate_station,
+					/area/skipjack_station,
+					/area/solar,
+					/area/shuttle,
+					/area/holodeck,
+					/area/supply/station,
+					/area/mine,
+					/area/vacant/vacant_shop
+					)
+
+	var/list/exempt_from_atmos = typesof(/area/maintenance,
+						/area/storage,
+						/area/engineering/atmos/storage,
+						/area/rnd/test_area,
+						/area/construction,
+						/area/server,
+						/area/mine,
+						/area/vacant/vacant_shop,
+						/area/rnd/research_storage, // This should probably be fixed,
+						/area/security/riot_control // This should probably be fixed,
+						)
+
+	var/list/exempt_from_apc = typesof(/area/construction,
+						/area/medical/genetics,
+						/area/mine,
+						/area/vacant/vacant_shop
+						)
+
+	for(var/area/A in world)
+		if(A.z == 1 && !(A.type in exempt_areas))
+			area_test_count++
+			var/area_good = 1
+			var/bad_msg = "--------------- [A.name]([A.type])"
+
+
+			if(isnull(A.apc) && !(A.type in exempt_from_apc))
+				log_unit_test("[bad_msg] lacks an APC.")
+				area_good = 0
+
+			if(!A.air_scrub_info.len && !(A.type in exempt_from_atmos))
+				log_unit_test("[bad_msg] lacks an Air scrubber.")
+				area_good = 0
+
+			if(!A.air_vent_info.len && !(A.type in exempt_from_atmos))
+				log_unit_test("[bad_msg] lacks an Air vent.")
+				area_good = 0
+		
+			if(!area_good)
+				bad_areas.Add(A)
+
+	if(bad_areas.len)
+		fail("\[[bad_areas.len]/[area_test_count]\]Some areas lacked APCs, Air Scrubbers, or Air vents.")
```

Comment:
Yeah i can see that, I think I made it a list so it could loop through the list at the end and report, I wound up instead reporting failures while in the main loop of the test.

Not sure why I left it like that, I'm pretty sure I had something else intended but I don't have my notes with me at work.